code/modules/mob/living/simple_animal/friendly/pet.dm

```diff
@@ -4,21 +4,23 @@
 	var/obj/item/clothing/neck/petcollar/pcollar = null
 	var/collar = ""
 	var/pettag = ""
+	var/can_wear_pet_items = TRUE
 	blood_volume = BLOOD_VOLUME_NORMAL
 
 /mob/living/simple_animal/pet/attackby(obj/item/O, mob/user, params)
 	if(istype(O, /obj/item/clothing/neck/petcollar) && !pcollar)
-		var/obj/item/clothing/neck/petcollar/P = O
-		pcollar = P
-		collar = "[icon_state]collar"
-		pettag = "[icon_state]tag"
-		regenerate_icons()
-		to_chat(user, "<span class='notice'>You put the [P] around [src]'s neck.</span>")
-		if(P.tagname)
-			real_name = "\proper [P.tagname]"
-			name = real_name
-		qdel(P)
-		return
+		if(can_wear_pet_items)
```

Comment:
Unnest this.

Reply:
Not immediately obvious what you want un-nested , are you referring to if(P.tagname) ?
Should only apply a name if the pet can receive a collar.